oracle/ojdbc-extensions
# Added support for TCPS and SEPS Wallet Providers for OCI Vault

this pull request adds support for TCPS and SEPS Wallet Providers for managing secure connections and credentials via OCI Vault in Oracle JDBC.

**Changes Introduced:**

        - TCPS Wallet Provider: This feature allows users to retrieve and manage Base64-encoded wallets (both PKCS12 and SSO) stored in OCI Vault. It enables secure mutual TLS (mTLS) communication by retrieving the TCPS/TLS wallet directly from the OCI Vault.

        - SEPS Wallet Provider: This feature supports retrieving SEPS wallets from OCI Vault. It extracts the username and password from the wallet to securely manage database credentials.

**How to Test:**

          - Set up the configuration file (e.g., example-vault.properties) with your wallet's OCID and, if applicable, the wallet password.

          - Use the new TCPS and SEPS wallet providers to validate connections:

                  1. For TCPS/TLS, ensure the wallet is retrieved from the OCI Vault and used for secure TLS communication.

                  2. For SEPS, verify that the wallet is retrieved from the OCI Vault, and the username and password are extracted from it.

Fixes Issue: #95
